lightfm/lightfm.py

```
@@ -833,9 +1351,9 @@ def predict(
                 f" of item IDs ({len(item_ids)})"
             )
 
-        if user_ids.dtype != np.int32:
+        if hasattr(user_ids, 'dtype') and user_ids.dtype != np.int32:
```

Ruff (0.11.9)

Undefined name `np`

(F821)